plugins/extract/align/_base.py

```diff
@@ -183,9 +183,14 @@ def finalize(self, batch):
 
         """
 
-        for face, landmarks in zip(batch["detected_faces"], batch["landmarks"]):
+        generator = zip(batch["detected_faces"],
+                        batch["filename"],
+                        batch["image"],
+                        batch["landmarks"])
+        for face, filename, image, landmarks in generator:
             face.landmarks_xy = [(int(round(pt[0])), int(round(pt[1]))) for pt in landmarks]
-
+            face.image = image
+            face.filename = filename
         self._remove_invalid_keys(batch, ("detected_faces", "filename", "image"))
```

Comment:
Not strictly necessary now. Believe there should be future modifications to pass a Detected Face class through all extract models instead of a dict of images, filenames, landmarks, Detected Face... especially considering each of those items are present inside Detected Face